### ISSUE TITLE
create power control scheme using VN7050ASTR MOSFET

### ISSUE THREAD

**Issue by nerdoug** [va3wam/hexBot#192]
In pursuing how to implement high side switching for 5V power control, I came across the STM VN7050ASTR chip which has a lot going for it:

On Thu, Jun 2, 2022 at 6:51 PM Doug Elliott <[canoe.eh@gmail.com](mailto:canoe.eh@gmail.com)> wrote:
This seems too good to be true:
  https://www.mouser.ca/ProductDetail/STMicroelectronics/VN7050ASTR?qs=N2Tevpb%252BvoaSglANpOhF3A%3D%3D

- high side switch
- up to 40 V
- up to 30 A
- control signals compatible with 3 and 5 Volt systems ( no level converters or Darlington's needed)
- built in current measurement
- 8 pin SO-9 package - surface mount, for 1 switched channel
- (need to verify that current sensing works on 8 pin part that's missing some SEL pins)
- in stock at Mouser
- $2.68 CDN each

I've since verified that the 8 pin part does support current sensing. They did reduce the functionality on the 8 pin part, but they maintained the ability to measure current:

Reference: ST Microelectronics  Application note UM1922 (attached)
- section 5
- table 15
- page 59
[um1922-vipower-m07-standard-highside-drivers-hardware-design-guide-stmicroelectronics.pdf](https://github.com/va3wam/hexBot/files/8836645/um1922-vipower-m07-standard-highside-drivers-hardware-design-guide-stmicroelectronics.pdf)

We still need to drive SELn High to get the current measurement feedback, but we could strap it rather than use a GPIO.

We could use this chip for all of our software controlled circuits:
- 5V enable, for both PCA9685's
- 3.3V enable, for both PCA9685's
- enable/disable the Battery Voltage measuring circuit
- enable/disable the 5V voltage measuring system.

The purpose of this issue is to create a power diagram version that uses the VN7050ASTR for all of these functions, for discussion purposes.

**Comment by nerdoug** [comment]
Snag: The circuit that the VN7050ASTR controls is specified as having a voltage above 4 volts, which means we can't use it to control the 3.3V circuit to the PCA9685's.  There's probably another chip variation we can use. Some possibilities:
    STMPS2242, 2252, 2262, 2272
    STMPS2141....

**Comment by nerdoug** [comment]
I have ordered and received a bunch of VN7050ASTR chips for 5V enable/disable, and current reading.
I also got a bunch of STMPS2141AMTTR chips to control the 3.3 V feed to the PCA9685's and to enable/disable the voltage divider used to measure battery voltage.

I done some breadboard testing, and they operate as expected with one exception: The sense of the EN pin on the 2141 is opposite to the 7050. On the 2141, if you put a logic high on the EN pin, it OPENS the switch. On the 7050, a logic high on INPUT CLOSES the switch. I don't think this is a problem, and we just need to adjust software, pullup/down resistors, and possibly the choice of GPIOs.

This behaviour doesn't seem to be mentioned in the 2141 data sheet, but it's implied in the diagrams in sections 7.1.1 and 7.1.2.

I've also made up a cheat sheet that shows how the chip pins are connected to the DIP carrier's pins, with a description of the pin function.  ..doc/circuitry/soic-8-adapters-pinouts.odg and pdf.
